### PR TITLE
connect-1486 : Add nginx config for nuxt

### PR DESCRIPTION
**Pull request by npiron**
## Change log
 - Mise en place de la configuration NgInx pour les projets Nuxt

 **PRs liées :**
 - https://github.com/ArDeveloppement/customizer-client/pull/479
 - https://github.com/ArDeveloppement/Arconnect-Deployment/pull/144

### CODE REVIEW

**Comment by nervo** [comment]
Please, add a test case :)
See: https://github.com/ArDeveloppement/docker-image-system/tree/master/tests/app

**Comment by nervo** [comment]
And don't forget to update the changelog, some packages like php or nginx could have been updated :)

**Comment by npiron** [comment]
@nervo, I just added some tests.
But we have trouble to test a Nuxt application.
Nuxt is just a reverse proxy witch redirect to a NodeJS server (Nuxt SSR).
I excluded nuxt from one test until we find a solution

If you have any advice, I'm interested :)

File: .gitignore [hunk 1]
```diff
@@ -1 +1 @@
-
+.idea/
```
Comment (nervo) [inline .gitignore]: 'should be part of your local .gitignore, not project's one, btw :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,11 +6,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.2.3] - 2019-10-15
+### Added
+- Handle Nuxt application
+    - Add nginx configuration
+    - Add `nuxt-start` in supervisor
+
```
Comment (nervo) [inline CHANGELOG.md]: Don't forget the php/node/... version updates part !

File: tests/app/goss.yaml [hunk 3]
```diff
@@ -18,11 +18,12 @@ http:
...
   http://localhost:
     status: 200
-{{ if .Vars.app }}
+  {{ if .Vars.app }}
     body:
       - '/^app: {{ .Vars.app }}$/'
-{{ end }}
-
+  {{ end }}
+{{ end }}
```
Comment (nervo) [inline tests/app/goss.yaml]: Missing trailing line

File: tests/app/goss.yaml [hunk 4]
```diff
@@ -18,11 +18,12 @@ http:
       - '/^pool: /'
       - '/^process manager: /'
 {{ end }}
+{{ if (not (eq (or .Vars.app "") "nuxt"))}}
   # App
   http://localhost:
     status: 200
-{{ if .Vars.app }}
+  {{ if .Vars.app }}
```
Comment (nervo) [inline tests/app/goss.yaml]: Are you sure there are no indentation issues doing this ?
Maybe something like `{{-‘ should handle them, if any...

File: etc/supervisor/app.conf.tpl [hunk 5]
```diff
@@ -21,3 +21,15 @@ stdout_logfile_maxbytes = 0
 stderr_logfile = /dev/stderr
 stderr_logfile_maxbytes = 0
 {{- end }}
+
+{{- if (and (eq (getenv "APP") "nuxt") (eq (getenv "ENVIRONMENT") "production"))}}
+[program:nuxt]
+directory=/srv/app/
+command=nuxt-start
```
Comment (nervo) [inline etc/supervisor/app.conf.tpl]: what exactly is `nuxt-start` ? A script ? A js file ?
As far as i can see, command should be something like "node foo.js" or "npx foo"

**Comment by nervo** [comment]
I think you have to simulate "something" running on port 3000, in a supervisor command compatible manner, responding a simple string easily testable.
This could be made in vanilla js, i guess...

File: .gitignore (hunk 1 above)
Comment (Spoon4) [reply .gitignore]: yes, we currently add IDE config to global ignore list. You can find it in the [documentation](https://app.gitbook.com/@cospirit-connect/s/guide-de-demarrage/installation-des-projets/prerequis/configuration-globale-git).

**Comment by npiron** [comment]
As mentioned, I create a fake server in NodeJS to simulate Nuxt and then I can test the reverse proxy.

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,11 +6,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [1.2.3] - 2019-10-15
+## [1.3.0] - 2019-10-15
 ### Added
 - Handle Nuxt application
     - Add nginx configuration
     - Add `nuxt-start` in supervisor
+- Update Php 7.2.23
```
Comment (nervo) [inline CHANGELOG.md]: `### Changed`  :)

**Pull request merged**